Data source: http://www.tse.com.tw/zh/page/trading/exchange/MI_5MINS_INDEX.html
1. before 2011-01-16 is a data per min
2. before 2014-02-23 is a data per 15 sec
3. before 2014-12-29 is a data per 10 sec
4. after 2014-12-29 to today is a data per 5 sec

What we need to complete this shit
1. script1: from 2004-10-15 to today
2. script2: from today keep update data, need to use another script
3. creat a DB with a table ontain every columns, different date have different column lenth. (how to make it auto variable with data change?)  (idea: for script1, just drop the fix numbers columns. Then, as script2 insert data, just insert the columns which has time column)  
4. drop the rows whose content is string, then insert date for every data .


In [2]:
import datetime as dt
import requests
import os
import time


def creat_folder():
    os.makedirs('./five_sec/', exist_ok=True)    
#   creat a forder in local to keep all file. As we creat once, we dont need it anymore.


def date_trans(start1,end1):

    date_int={'year1':int(start1[:4]),'month1':int(start1[4:6]),'day1':int(start1[6:]),
              'year2':int(end1[:4]),'month2':int(end1[4:6]),'day2':int(end1[6:])}
    
    start_date = dt.datetime(date_int['year1'],date_int['month1'],date_int['day1'])
    end_date = dt.datetime(date_int['year2'],date_int['month2'],date_int['day2'])
    totaldays = (end_date - start_date).days + 1
    return start_date,totaldays


def get_content(start_date,totaldays):
    
    for daynumber in range(totaldays):
        date1=(start_date + dt.timedelta(days = daynumber)).date()

        five_sec_url='http://www.tse.com.tw/exchangeReport/MI_5MINS_INDEX?response=csv&date=%s'%str(date1)
        r = requests.get(five_sec_url)
        
        lst = split_content(r)
        print(lst)
#         print(len(lst))   #use to check whether the size of file is same or not 

        for j in range(len(lst)):  # range should be change to len(lst), but now just for test the sample
            if j>3:                # first two line is title, so we skip it
                tup = [parser(i)  for i in lst[j].split(',')]     
                # should add ome more condition to control title and the last text 
                
#                 print(tup)
#         download_file(r,date1)
 

# def download_file(r,date1):
# this is use to download file, while we don't need now.
#     if date1.weekday() < 5:
#         with open('./five_sec/%s.csv'@str(date1),'wb')as f:
#             f.write(r.content)


def parser(s):
    s = s.strip('""').replace(",", "")
    if ":" not in s and len(s)>0:
        s = float(s)
    return s


def split_content(r):
    s = str(r.content)
    lst = s.split("\\r\\n")
    return lst


# main
# creat_file()
t1 = time.time()  # record start time

start_date,totaldays=date_trans('20180810','20180810')
get_content(start_date,totaldays)

print('Total time:',time.time()-t1)    # count total time

['b\'"106\\xa6~12\\xa4\\xeb08\\xa4\\xe9\\xa8C5\\xac\\xed\\xab\\xfc\\xbc\\xc6\\xb2\\xce\\xadp"', '"\\xae\\xc9\\xb6\\xa1","\\xb5o\\xa6\\xe6\\xb6q\\xa5[\\xc5v\\xaa\\xd1\\xbb\\xf9\\xab\\xfc\\xbc\\xc6","\\xa5\\xbc\\xa7t\\xaa\\xf7\\xbf\\xc4\\xabO\\xc0I\\xaa\\xd1\\xab\\xfc\\xbc\\xc6","\\xa5\\xbc\\xa7t\\xb9q\\xa4l\\xaa\\xd1\\xab\\xfc\\xbc\\xc6","\\xa5\\xbc\\xa7t\\xaa\\xf7\\xbf\\xc4\\xb9q\\xa4l\\xaa\\xd1\\xab\\xfc\\xbc\\xc6","\\xa4\\xf4\\xaad\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xad\\xb9\\xab~\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xb6\\xec\\xbd\\xa6\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xaf\\xbc\\xc2\\xb4\\xc5\\xd6\\xba\\xfb\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xb9q\\xbe\\xf7\\xbe\\xf7\\xb1\\xf1\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xb9q\\xbe\\xb9\\xb9q\\xc6l\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xa4\\xc6\\xbe\\xc7\\xa5\\xcd\\xa7\\xde\\xc2\\xe5\\xc0\\xf8\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xa4\\xc6\\xbe\\xc7\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xa5\\xcd\\xa7\\xde\\xc2\\xe5\\xc0\\xf8\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xac\

ValueError: could not convert string to float: '\\xa5\\xc1\\xb0\\xea100\\xa6~1\\xa4\\xeb16\\xa4\\xe9\\xa5H\\xabe\\xa1A\\xb8\\xea\\xae\\xc6\\xac\\xb0\\xa8C\\xa4@\\xa4\\xc0\\xc4\\xc1\\xa4\\xe8\\xa6\\xa1\\xb4\\xa3\\xa8\\xd1\\xa1C'

In [2]:
# lst_=[]
# for i in lst[-10].split(','):
#     ele = i.strip('""').replace(",", "")
#     print(ele)
#     if ":" not in ele and len(ele)>0:
#         if ele[0].isdigit():
#             ele = float(ele)
#     lst_ += [ele]   
# print(lst_)

['b\'"106\\xa6~12\\xa4\\xeb08\\xa4\\xe9\\xa8C5\\xac\\xed\\xab\\xfc\\xbc\\xc6\\xb2\\xce\\xadp"', '"\\xae\\xc9\\xb6\\xa1","\\xb5o\\xa6\\xe6\\xb6q\\xa5[\\xc5v\\xaa\\xd1\\xbb\\xf9\\xab\\xfc\\xbc\\xc6","\\xa5\\xbc\\xa7t\\xaa\\xf7\\xbf\\xc4\\xabO\\xc0I\\xaa\\xd1\\xab\\xfc\\xbc\\xc6","\\xa5\\xbc\\xa7t\\xb9q\\xa4l\\xaa\\xd1\\xab\\xfc\\xbc\\xc6","\\xa5\\xbc\\xa7t\\xaa\\xf7\\xbf\\xc4\\xb9q\\xa4l\\xaa\\xd1\\xab\\xfc\\xbc\\xc6","\\xa4\\xf4\\xaad\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xad\\xb9\\xab~\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xb6\\xec\\xbd\\xa6\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xaf\\xbc\\xc2\\xb4\\xc5\\xd6\\xba\\xfb\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xb9q\\xbe\\xf7\\xbe\\xf7\\xb1\\xf1\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xb9q\\xbe\\xb9\\xb9q\\xc6l\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xa4\\xc6\\xbe\\xc7\\xa5\\xcd\\xa7\\xde\\xc2\\xe5\\xc0\\xf8\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xa4\\xc6\\xbe\\xc7\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xa5\\xcd\\xa7\\xde\\xc2\\xe5\\xc0\\xf8\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xac\\xc1\\xbc\\xfe\\xb3\\xb3\\xb2\\xa1\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xb3y\\xaf\\xc8\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xbf\\xfb\\xc5K\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xbe\\xf3\\xbd\\xa6\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xa8T\\xa8\\xae\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xb9q\\xa4l\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xa5b\\xbe\\xc9\\xc5\\xe9\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xb9q\\xb8\\xa3\\xa4\\xce\\xb6g\\xc3\\xe4\\xb3]\\xb3\\xc6\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xa5\\xfa\\xb9q\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xb3q\\xabH\\xba\\xf4\\xb8\\xf4\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xb9q\\xa4l\\xb9s\\xb2\\xd5\\xa5\\xf3\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xb9q\\xa4l\\xb3q\\xb8\\xf4\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xb8\\xea\\xb0T\\xaaA\\xb0\\xc8\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xa8\\xe4\\xa5L\\xb9q\\xa4l\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xab\\xd8\\xa7\\xf7\\xc0\\xe7\\xb3y\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xaf\\xe8\\xb9B\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xc6[\\xa5\\xfa\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xaa\\xf7\\xbf\\xc4\\xabO\\xc0I\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xb6T\\xa9\\xf6\\xa6\\xca\\xb3f\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xaao\\xb9q\\xbfU\\xae\\xf0\\xc3\\xfe\\xab\\xfc\\xbc\\xc6","\\xa8\\xe4\\xa5L\\xc3\\xfe\\xab\\xfc\\xbc\\xc6",', 